compiler/crates/relay-compiler/src/build_project/mod.rs

```diff
@@ -373,7 +382,9 @@ pub async fn commit_project(
                     break;
                 }
                 let path = config.root_dir.join(remaining_artifact);
-                config.artifact_writer.remove(path)?;
+                if is_relay_file_path(&project_config.typegen_config.language, &path) {
```

Comment:
This is mostly nits (and not sure if it makes it better). But can we move this function to the artifact_writer?

We already have `should_write`: https://github.com/facebook/relay/blob/e72ea12c746735f8bfa8e59e1c9704fcd6cd2179/compiler/crates/relay-compiler/src/build_project/artifact_writer.rs#L21

I think we can also add `should_remove`?

Reply:
`should_remove` can just return a boolean instead of `Result<...>`.

Reply:
I'd need to make sure that the typegen config is added to the artifact_writer if so, but maybe it's ok to add that? Or I could do a typegen unaware check and just check for .graphql.js or .graphql.ts always. What do you think?

Reply:
I think it's fine to do a check without typegen config.

What do you think about alternative approach, where we would filter these non-relay artifacts in the `file_categorizer`?

Somewhere here: https://github.com/facebook/relay/blob/a07750f252e409e419e6886ec4a3d2a20cc4c48c/compiler/crates/relay-compiler/src/file_source/file_categorizer.rs#L237-L239